### ISSUE TITLE
Directory timestamps are not hashed

### ISSUE THREAD

**Issue by mwaskom** [nipy/nipype#638]
When an interface input is a directory (I encountered this with DataSink; it may be specific to that interface), it is treated as a string and only the name is used to generate the hash. This is problematic when the contents of the directory, but not the directory name, change. I can't track down the root code that does the hashing, but I can only assume that there is an 

```
if os.path.isfile(s):
    ... hash file info
else:
    ... hash name
```

somewhere. It should be easy to add a directory check onto this code, I just don't know where it is. I'm also not totally sure what the right behavior should be when the config option for hashing says to use file contents.

**Comment by djarecka** [comment]
@satra : didn't your PR #646 (changes in `io.py`) solve the issue? can we close it?

**Comment by djarecka** [comment]
@satra ping

**Comment by satra** [comment]
@djarecka - yup that fixes this issue. fixed by #646
